launch/config.py

```diff
@@ -0,0 +1,393 @@
+    assert int(launch_config_version) == 1
+
+
+def validate_onprem_dcos_config_contents(onprem_dcos_config_contents, num_masters):
+    """ TODO: Remove conditional logic here by placing this function in a
+    conditional entry. At present, all validates are global. See: DCOS-14033
+    """
+    user_config = yaml.load(onprem_dcos_config_contents)
+    # Use the default config in the installer
+    config = yaml.load(dcos_installer.config.config_sample)
+    config.update(user_config)
+    # This field is required and auto-added by installer, so add a dummy here
+    if 'bootstrap_id' not in config:
+        config['bootstrap_id'] = 'deadbeef'
+
+    # Error message will instruct user to provide ip_detect_contents if we dont
+    # have the filename argument provided. We want users providing the file
+    if 'ip_detect_filename' not in config:
+        config['ip_detect_filename'] = 'provide-this-ip-detect-path'
+
+    # dummy master list to pass validation
+    config['master_list'] = [('10.0.0.' + str(i)) for i in range(int(num_masters))]
+
+    # Use the default config in the installer
+    sources, targets, templates = gen.get_dcosconfig_source_target_and_templates(
+        gen.stringify_configuration(config), list(), [ssh.validate.source, gen.build_deploy.bash.onprem_source])
+
+    # Copy the gen target from dcos_installer/config.py, but instead remove
+    # 'ssh_key_path' from the target because the validate fn in ssh_source is
+    # too strict I.E. we cannot validate a key if we are going to generate
+    # Furthermore, we cannot use the target ssh_key_path as it will automatically
+    # invoked the validate fn from ssh/validate.py Luckily, we can instead use
+    # the more idiomatic 'ssh_private_key_filename'
+    targets.append(Target({
+        'ssh_user',
+        'ssh_port',
+        # 'ssh_key_path,
+        'master_list',
+        'agent_list',
+        'public_agent_list',
+        'ssh_parallelism',
+        'process_timeout'}))
+
+    resolver = resolve_configuration(sources, targets)
+    status = resolver.status_dict
+    if status['status'] == 'errors':
+        raise AssertionError(pretty_print_validate_error(status['errors'], status['unset']))
+
+
+def calculate_dcos_config_contents(dcos_config, num_masters, ssh_user):
+    user_config = yaml.load(dcos_config)
+    # Use the default config in the installer for the same experience
+    # w.r.t the auto-filled settings
+    config = yaml.load(dcos_installer.config.config_sample)
+    config.update(user_config)
+    config['ssh_user'] = ssh_user
+    return yaml.dump(config)
+
+
+def validate_provider_platform(provider, platform):
+    """ TODO: Remove conditional logic here by placing this function in a
+    conditional entry. At present, all validates are global. See: DCOS-14033
+    """
+    if platform in ('aws', 'azure'):
+        assert platform != 'bare_cluster', 'bare_cluster only supports "provider: onprem"'
```

Comment:
onprem is never a platform. bare_cluster is a platform that supports onprem, nonetheless, I see now that this validate is a bit of non-sense and will fix